obspy/core/inventory/response.py

```diff
@@ -1243,6 +1362,39 @@ def get_unit_mapping(key):
             if blkt is not None:
                 stage_blkts.append(blkt)
 
+            # Evalresp requires FIR and IIR blockettes to have decimation
+            # values. Set the "unit decimation" values in case they are not
+            # set.
+            #
+            # Only set it if there is a stage gain - otherwise evalresp
+            # complains again.
+            if isinstance(blockette, PolesZerosResponseStage) and \
+                    blockette.stage_gain and \
+                    None in set([
+                        blockette.decimation_correction,
+                        blockette.decimation_delay,
+                        blockette.decimation_factor,
+                        blockette.decimation_input_sample_rate,
+                        blockette.decimation_offset]):
+                # Don't modify the original object.
+                blockette = copy.deepcopy(blockette)
+                blockette.decimation_correction = 0.0
+                blockette.decimation_delay = 0.0
+                blockette.decimation_factor = 1
+                blockette.decimation_offset = 0
+                sr = self.get_sampling_rates()
+                if sr and blockette.stage_sequence_number in sr and \
+                        sr[blockette.stage_sequence_number][
+                            "input_sampling_rate"]:
+                    blockette.decimation_input_sample_rate = \
```

Comment:
You don't wanna set the decimation_factor too?

Reply:
Yea I thought about this as well but if no decimation attribute is set it must be 1 in any case. Just seems a tiny bit safer in case there is some response file out there which manages to confuse the `get_sampling_rates()` method.